codalab/apps/api/tests/test_old_api.py

```diff
@@ -134,95 +161,103 @@ def setUp(self):
+                }
+            )
 
-        self.assertEquals(resp.status_code, 200)
+            self.assertEqual(resp.status_code, 200)
 
-        subjects = [m.subject for m in mail.outbox]
-        self.assertIn('Accepted into %s!' % self.competition, subjects)
-        self.assertIn('%s accepted into your competition!' % self.participant_user, subjects)
+            subjects = [m.subject for m in mail.outbox]
+            self.assertIn('Accepted into %s!' % self.competition, subjects)
+            self.assertIn('%s accepted into your competition!' % self.participant_user, subjects)
 
     def test_participation_status_update_revoked_sends_email(self):
-        self._participant_join_competition(cleanup_email=True)
+        with mock.patch.object(CompetitionAPIViewSet, '_send_mail', new=_new_send_mail):
+            self._participant_join_competition(cleanup_email=True)
 
-        participant = CompetitionParticipant.objects.get(competition=self.competition, user=self.participant_user)
+            participant = CompetitionParticipant.objects.get(competition=self.competition, user=self.participant_user)
 
-        self.client.login(username="organizer", password="pass")
-        resp = self.client.post(
-            reverse('competition-participation-status', kwargs={'pk': self.competition.pk}),
-            {
-                "status": "denied",
-                "participant_id": participant.pk,
-                "reason": ""
-            }
-        )
+            self.client.login(username="organizer", password="pass")
+            resp = self.client.post(
+                reverse('competition-participation-status', kwargs={'pk': self.competition.pk}),
+                {
+                    "status": "denied",
+                    "participant_id": participant.pk,
+                    "reason": ""
+                }
+            )
 
-        self.assertEquals(resp.status_code, 200)
+            self.assertEqual(resp.status_code, 200)
 
-        subjects = [m.subject for m in mail.outbox]
-        self.assertIn('Permission revoked from Test Competition!', subjects)
-        self.assertIn("%s's permission revoked from your competition!" % self.participant_user, subjects)
+            subjects = [m.subject for m in mail.outbox]
+            self.assertIn('Permission revoked from Test Competition!', subjects)
+            self.assertIn("%s's permission revoked from your competition!" % self.participant_user, subjects)
 
     def test_participation_status_update_not_sent_when_participant_disables_status_notifications(self):
-        self.participant_user.participation_status_updates = False
-        self.participant_user.save()
+        with mock.patch.object(CompetitionAPIViewSet, '_send_mail', new=_new_send_mail):
+            self.participant_user.participation_status_updates = False
+            self.participant_user.save()
 
-        self._participant_join_competition()
+            self._participant_join_competition()
```

Comment:
why did these get moved inside the mock? we should have as little in the mocks as possible so their context is clear?

Reply:
`self._participant_join_competition()` needs to be in the mock since it hits the API point targeted by it.

I've moved everything else outside of the mock to make it more clear.

Reply:
I'm a little confused, previously this test wasn't passing or something?

Reply:
Yes, the mock was necessary to fix emails not being sent in tests for later Django versions when you don't directly call `send_mail` or call the `.send()` method on Email objects.

The mock just grabs the email backend and forces it to send emails it receives. There should be a comment above the mock as well to explain the reasoning.